lib/wikipedia-transform.js

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const EXPORTED_SYMBOLS = ['wikipediaTransform'];
+
+const IMAGE_DIMENSION = 32;
+
+function WikipediaTransform() {
+  /*
+  Class that normalizes the wikipedia type API responses into a data object
+  containing all required keys. Used by the Recommendation to generate a
+  uniform data object to pass into the RecommendationRow for rendering into the
+  XUL DOM.
+
+  This class is stateless, so a singleton instance is created and exported.
+  */
+}
+WikipediaTransform.prototype = {
+  transform: function(data) {
+    if (!this.isValid(data)) {
+      return;
+    }
+    let dimensions = {};
+    if (data.enhancements.wikipedia.image.height &&
+        data.enhancements.wikipedia.image.width) {
+      dimensions = this.calculateDimensions(data);
+    }
+    // NOTE: using the full keys repeatedly to aid grepping.
+    let result = {
+      type: 'wikipedia',
+      favicon: {
+        url: 'chrome://universalsearch-skin/content/wikipedia-dark.svg',
+        color: data.enhancements.favicon.color
+      },
+      title: data.enhancements.wikipedia.title,
+      url: data.enhancements.wikipedia.url,
+      keyImage: {
+        url: data.enhancements.wikipedia.image.url,
+        height: dimensions.height,
```

Comment:
It assumes truthiness and checks if the image URL exists.

So, the way to avoid rendering some broken junk is to append the keyImage after `result` is created, and only if the url is set.